tools/cloud_functions/gcs_event_based_ingest/gcs_ocn_bq_ingest/common/utils.py

```diff
@@ -40,96 +42,157 @@
-    an external table definition _config/external.json (otherwise will assume
-    CSV external table)
+    an external table definition
+    _config/{constants.BQ_EXTERNAL_TABLE_CONFIG_FILENAME} (otherwise will assume
+    PARQUET external table)
     """
     external_table_config = read_gcs_file_if_exists(
-        gcs_client, f"{gsurl}_config/external.json")
+        gcs_client,
+        f"{gsurl}_config/{constants.BQ_EXTERNAL_TABLE_CONFIG_FILENAME}")
     if not external_table_config:
         external_table_config = look_for_config_in_parents(
-            gcs_client, gsurl, "external.json")
+            gcs_client, gsurl, constants.BQ_EXTERNAL_TABLE_CONFIG_FILENAME)
     if external_table_config:
         external_table_def = json.loads(external_table_config)
     else:
-        print(f" {gsurl}_config/external.json not found in parents of {gsurl}. "
+        print(f" {gsurl}_config/{constants.BQ_EXTERNAL_TABLE_CONFIG_FILENAME} "
+              f"not found in parents of {gsurl}. "
               "Falling back to default PARQUET external table: "
               f"{json.dumps(constants.DEFAULT_EXTERNAL_TABLE_DEFINITION)}")
         external_table_def = constants.DEFAULT_EXTERNAL_TABLE_DEFINITION
-
+    print(
+        json.dumps(
+            dict(message="Found external table definition.",
+                 table=table.to_api_repr(),
+                 external_table_def=external_table_def)))
+    # Reduce the amount of sourceUris by using wildcards with common
+    # prefixes. This is done to keep the cloud logging audit metadata
+    # below 100k in size, otherwise the metadata is omitted in the event.
+    source_uris_with_wildcards = compact_source_uris_with_wildcards(
+        flatten2dlist(get_batches_for_gsurl(gcs_client, gsurl)))
     # This may cause an issue if >10,000 files.
-    external_table_def["sourceUris"] = flatten2dlist(
-        get_batches_for_prefix(gcs_client, gsurl))
-    print(f"external table def = {json.dumps(external_table_config, indent=2)}")
+    external_table_def["sourceUris"] = source_uris_with_wildcards
+    # Check if hivePartitioningOptions
+    # https://cloud.google.com/bigquery/docs/reference/rest/v2/tables#hivepartitioningoptions
+    # is set in external.json file
+    if external_table_def.get("hivePartitioningOptions"):
+        external_table_def["hivePartitioningOptions"] = {
+            "mode":
+                external_table_def["hivePartitioningOptions"].get("mode")
+                or "AUTO",
+            "sourceUriPrefix":
+                get_hive_partitioning_source_uri_prefix(
+                    external_table_def["sourceUris"][0])
+        }
     external_config = bigquery.ExternalConfig.from_api_repr(external_table_def)
     job_config = bigquery.QueryJobConfig(
         table_definitions={"temp_ext": external_config}, use_legacy_sql=False)
 
-    # drop partition decorator if present.
-    table_id = dest_table_ref.table_id.split("$")[0]
+    if table:
```

Comment:
This validation confuses me a bit. (I might be missing something).

I guess we are trying to check if `table is None`.

1) If `table is None` I believe the CF might exit without writing any data to BQ, isn't this worthy of being logged or raising a warning? Is there a case where this is the desired behavior?

2) In this same function, a method is called on the `table` variable previous to this validation. Would that not trigger a an error if `table is None`?

Reply:
This is a good catch, I think we can just eliminate the 'If' check.

Reply:
Agree